### PR TITLE
fix(swap): allow one extrac block to be created during swap

### PR DESCRIPTION
**Pull request by offerm**

### CODE REVIEW

File: lib/swaps/Swaps.ts [hunk 1]
```diff
@@ -569,7 +569,8 @@ class Swaps extends EventEmitter {
       return false;
     }
 
-    if (cltvDelta > resolveRequest.getTimeout() - resolveRequest.getHeightNow()) {
+    // allow 1 additional one block to be created during the swap
+    if (cltvDelta - 1 > resolveRequest.getTimeout() - resolveRequest.getHeightNow()) {
       this.logger.error(`got timeout ${resolveRequest.getTimeout()} at height ${resolveRequest.getHeightNow()}`);
       this.logger.error(`cltvDelta is ${resolveRequest.getTimeout() - resolveRequest.getHeightNow()} expected delta of ${cltvDelta}`);
```
Comment (sangaman) [inline lib/swaps/Swaps.ts]: I figure this should say `expected delta of at least ${cltvDelta -1}`.

Comment (offerm) [reply lib/swaps/Swaps.ts]: I don't think we should change this. If we do it may create confusion with the configuration.

**Comment by offerm** [comment]
I will continue thinking about this. Not simple. 
Still, 2 new blocks (same net) during the swap is very unlikely. 

@sangaman Lets accept it as is since I need this for the trading tests.

Note that I added ceil in the ctlv calculation to avoid fraction (have seen this in action).

**Comment by sangaman** [comment]
Alright sounds good for now, thanks.

**Pull request merged**